llvm/lib/CodeGen/TargetPassConfig.cpp

```diff
@@ -1067,8 +1073,21 @@ bool TargetPassConfig::addISelPasses() {
   addPassesToHandleExceptions();
   if (YkBlockDisambiguate)
     addPass(createYkBlockDisambiguatePass());
-  addISelPrepare();
 
+  // We insert the yk control point pass as late as possible. It has to run
+  // before instruction selection (or the machine IR won't reflect our
+  // patching), but after other passes which mutate the IR (e.g.
+  // `CodegenPrepare` above).
+  //
+  // By running the pass late, we also ensure that no IR optimisation passes
+  // ever see our patched-in control point function. This would be problematic,
```

Comment:
s/This would be problematic/If optimisations were to be applied...

Just to make it clear that running this pass late isn't what's being problematic.

Reply:
Fixed in f9a7adc96a53